File: app/src/main/java/com/bth/running/core/RunningApp.java
package com.bth.running.core;

import android.app.Application;

import com.bth.running.dagger.AppComponent;
import com.bth.running.dagger.AppModule;
import com.bth.running.dagger.DaggerAppComponent;

import net.danlew.android.joda.JodaTimeAndroid;

import io.realm.Realm;

/**
 * @author Martin Macheiner
 *         Date: 05.09.2017.
 */

public class RunningApp extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        Realm.init(this);
        JodaTimeAndroid.init(this);

        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }


}

File: app/src/main/java/at/shockbytes/corey/running/core/RunningApp.kt
package at.shockbytes.corey.running.core

import android.app.Application
import at.shockbytes.corey.running.dagger.AppComponent
import at.shockbytes.corey.running.dagger.AppModule
import at.shockbytes.corey.running.dagger.DaggerAppComponent
import at.shockbytes.corey.running.dagger.NetworkModule
import net.danlew.android.joda.JodaTimeAndroid

/**
 * @author  Martin Macheiner
 * Date:    05.09.2017.
 */

class RunningApp : Application() {

    lateinit var appComponent: AppComponent
        private set

    override fun onCreate() {
        super.onCreate()

        JodaTimeAndroid.init(this)

        appComponent = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .networkModule(NetworkModule(this))
                .build()
    }


}
